src/cudadev/CondFormats/SiPixelROCsStatusAndMappingWrapper.h

```diff
@@ -28,17 +28,21 @@ class SiPixelROCsStatusAndMappingWrapper {
   std::vector<unsigned char, cms::cuda::HostAllocator<unsigned char>> modToUnpDefault;
   bool hasQuality_;
 
-  SiPixelROCsStatusAndMapping *cablingMapHost = nullptr;  // pointer to struct in CPU
+  cms::cuda::host::unique_ptr<std::byte[]> cablingMapHostBuffer;  // host pined memory for cabling map.
 
   struct GPUData {
-    ~GPUData();
-    SiPixelROCsStatusAndMapping *cablingMapDevice = nullptr;  // pointer to struct in GPU
+    void allocate(size_t size, cudaStream_t stream) {
+      cablingMapDeviceBuffer = cms::cuda::make_device_unique<std::byte[]>(
+              SiPixelROCsStatusAndMapping::computeDataSize(size), stream);
+      new(&cablingMapDevice) SiPixelROCsStatusAndMapping(cablingMapDeviceBuffer.get(), size);
```

Comment:
Should the destructor of earlier `calingMapDevice` object be called first?

Reply:
Not in this case, as the SoA is just a non-owning view with nothing to destroy, but I agree it's a good practice before an in-place constructor, and in case the class evolves. I will add it.